### DISCUSSION TITLE
Mix of query object and $queryRaw

### DISCUSSION THREAD

**Discussion by JanMalch** [prisma/prisma#18288] [Q&A] [answered]
Hi, is there some way to mix a query object with raw SQL? Or transform the query object to an SQL string?

I want to use SQLite's `ORDER BY RANDOM()`. I know there are suggestions with random skips, but I also need a random order and repeating the query x times (while tracking used indices to prevent duplicates) doesn't sound like a efficient way.
Unfortunately my query object can be quite complex. And whether or not random order is used, is determined by another parameter. So I have an `if` branch where I want use the query object (with a "normal" `orderBy` value) for `findMany`, and an `else` branch with the exact same query .. just with `ORDER BY RANDOM()`.

Hope it's clear what my thoughts and problems are. Thanks in advance!

**Comment by nurul3101** [comment, marked as answer]
Hello @JanMalch 👋 

You can [enable logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) to get the underlying SQL query from Prisma Client Query.

Also, we have a related feature request that might help in your use case
- https://github.com/prisma/prisma/issues/5848

**Reply by JanMalch** [reply]
Thanks, I managed to rewrite it in SQL.

Would _love_ to see some kind of random order support by Prisma, like proposed in the issue you linked. I'll keep an eye on that issue.

**Reply by nurul3101** [reply]
Thanks for getting back and confirming! 🙌
